src/plugins/minimal_scene/MinimalSceneRhiVulkan.cc

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include "MinimalSceneRhiVulkan.hh"
+
+#include "EngineToQtInterface.hh"
+#include "MinimalScene.hh"
+
+#include <gz/common/Console.hh>
+#include <gz/rendering/Camera.hh>
+
+#include <QMutex>
+#include <QQuickWindow>
+#include <QSGTexture>
+#include <QSize>
+
+#include <vulkan/vulkan_core.h>
+
+#include <memory>
+#include <string>
+
+/////////////////////////////////////////////////
+namespace gz
+{
+namespace gui
+{
+namespace plugins
+{
+  class GzCameraTextureRhiVulkanPrivate
+  {
+    public: VkImage textureId = 0;
+  };
+
+  class RenderThreadRhiVulkanPrivate
+  {
+    public: GzRenderer *renderer = nullptr;
+    public: void *texturePtr = nullptr;
+    public: QOffscreenSurface *surface = nullptr;
+  };
+
+  class TextureNodeRhiVulkanPrivate
+  {
+    public: VkImage textureId = 0;
+    public: VkImage newTextureId = 0;
+    public: QSize size {0, 0};
+    public: QSize newSize {0, 0};
+    public: QMutex mutex;
+    public: QSGTexture *texture = nullptr;
+    public: QQuickWindow *window = nullptr;
+  };
+}
+}
+}
+
+using namespace gz;
+using namespace gui;
+using namespace plugins;
+
+/////////////////////////////////////////////////
+GzCameraTextureRhiVulkan::~GzCameraTextureRhiVulkan() = default;
+
+/////////////////////////////////////////////////
+GzCameraTextureRhiVulkan::GzCameraTextureRhiVulkan()
+  : dataPtr(std::make_unique<GzCameraTextureRhiVulkanPrivate>())
+{
+}
+
+/////////////////////////////////////////////////
+void GzCameraTextureRhiVulkan::Update(rendering::CameraPtr _camera)
+{
+  // It says Metal but it also works for Vulkan in the exact same way
+  _camera->RenderTextureMetalId(&this->dataPtr->textureId);
+}
+
+/////////////////////////////////////////////////
+/////////////////////////////////////////////////
```

Comment:
```suggestion
```

Reply:
The double comment is on purpose (it's also in MinimalSceneRhiOpenGL.cc.cpp & MinimalSceneRhiMetal.mm) to distinguish the switch from `GzCameraTextureRhiVulkan` class definitions to `RenderThreadRhiVulkan`.

(fun fact: GzCameraTextureRhi* don't seem to actually be used. It seems to be dead code but this theory needs to be tested; I added a TODO in `class GzCameraTextureRhi`).